build.gradle

```diff
@@ -0,0 +1,103 @@
+/*
+ * This file was generated by the Gradle 'init' task.
+ *
+ * This generated file contains a sample Java application project to get you started.
+ * For more details take a look at the 'Building Java & JVM projects' chapter in the Gradle
+ * User Manual available at https://docs.gradle.org/7.5.1/userguide/building_java_projects.html
+ */
+
+plugins {
+    id 'groovy'
+    id 'java'
+    id 'jacoco'
+    id 'jacoco-report-aggregation'
+    id 'com.diffplug.spotless' version '6.11.0'
+    id 'com.github.johnrengelman.shadow' version '8.0.0'
+    id 'org.cyclonedx.bom' version '1.7.4'
+    id 'org.sonarqube' version '4.0.0.2929'
+}
+
+dependencies {
+    jacocoAggregation project(':app')
+    jacocoAggregation project(':shared')
+    jacocoAggregation project(':etor')
+}
+
+allprojects {
+    repositories {
+        mavenCentral()
+    }
+}
+
+subprojects {
+    apply plugin: 'org.sonarqube'
+    apply plugin: 'jacoco'
+    apply plugin: 'com.diffplug.spotless'
+    apply plugin: 'org.cyclonedx.bom'
+
+    cyclonedxBom {
+
+        // Specified the type of project being built. Defaults to 'library'
+        projectType = "application"
+
+        // Boms destination directory (defaults to build/reports)
+
+        // The file name for the generated BOMs (before the file format suffix). Defaults to 'bom'
+        outputName = "bom"
+
+        // The file format generated, can be xml, json or all for generating both
+        outputFormat = "all"
+
+        // Exclude BOM Serial Number
+        includeBomSerialNumber = false
+
+    }
+
+    jacoco {
+        toolVersion = '0.8.10'
+    }
+
+    spotless {
+        java {
+            googleJavaFormat().aosp()
+        }
+
+        groovy {
+            greclipse().configFile('../greclipse.properties')
+            excludeJava()
+        }
+
+        groovyGradle {
+            greclipse().configFile('../greclipse.properties')
+        }
+    }
+}
+
+ext.jacoco_excludes = [
+        '**/javalin/App*',
+        '**/jackson/Jackson*',
+        '**/slf4j/LocalLogger*',
+        '**/slf4j/DeployedLogger*',
+        '**/slf4j/LoggerHelper*',
+        '**/hapi/HapiFhirImplementation*',
+        '**/jjwt/JjwtEngine*',
+        '**/apache/ApacheClient*',
+        '**/azure/AzureSecrets*'
+]
+
+tasks.register('allUnitTests') {
```

Comment:
Could we have a similar task for all builds? e.g. `allBuilds`

Reply:
Are you thinking it would be used for the the Sonar scan?  In any case, sure, happy to.

Reply:
Yes, I thought about it after I saw the sonar scan build call, but it could also be useful as a shorthand